### PR TITLE
chore: Fix some elements in the .md that could cause confusion (repor…

### PR DESCRIPTION
**Pull request by kcelia**
- Fix some elements in the .md that could cause confusion (reported by a user).
    link: https://discord.com/channels/901152454077452399/1083021638544670720/1273533424187084800
-   Add a check to ensure the model is a PyTorch model, and a simple test.

### CODE REVIEW

File: docs/guides/hybrid-models.md [hunk 1]
```diff
@@ -141,7 +141,7 @@ For inference with the `HybridFHEModel` instance, `hybrid_model`, call the regul
 <!--pytest-codeblocks:skip-->
 
 ```python
-hybrid_model.forward(torch.randn((dim, )))
+hybrid_model(torch.randn((dim, )))
```
Comment (bcm-at-zama) [inline docs/guides/hybrid-models.md:144]: here, do we call the quantized module or really the simulate/fhe function?

File: docs/guides/hybrid-models.md [hunk 2]
```diff
@@ -141,7 +141,7 @@ For inference with the `HybridFHEModel` instance, `hybrid_model`, call the regul
...
 ```
 
 When calling `forward`, the `HybridFHEModel` handles all the necessary intermediate steps for each model part deployed remotely, including:
```
Comment (bcm-at-zama) [inline docs/guides/hybrid-models.md:147]: if the previous line is changed by removing the `forward`, then you would have to edit this line as well, which starts by

> When calling `forward`

File: src/concrete/ml/torch/hybrid_model.py [hunk 3]
```diff
@@ -343,21 +343,27 @@ class HybridFHEModel:
...
         self,
         model: nn.Module,
         module_names: Union[str, List[str]],
-        server_remote_address=None,
+        server_remote_address: Optional[str] = None,
         model_name: str = "model",
         verbose: int = 0,
     ):
+        if not isinstance(model, torch.nn.Module):
+            raise TypeError("The model must be a PyTorch model.")
```
Comment (bcm-at-zama) [inline src/concrete/ml/torch/hybrid_model.py:365]: in the doc, you had commented

>    model,  # PyTorch or Brevitas model

so:
- should the error string contain "Brevitas" as well
- is the `if not isinstance(model, torch.nn.Module):` valid for Brevitas case as well

File: tests/torch/test_hybrid_converter.py [hunk 4]
```diff
@@ -229,3 +227,14 @@ def test_gpt2_hybrid_mlp_module_not_found():
     fake_module_name = "does_not_exist"
     with pytest.raises(ValueError, match=f"No module found for name {fake_module_name}"):
         HybridFHEModel(model, module_names=fake_module_name)
+
+
+def test_invalid_model():
```
Comment (bcm-at-zama) [inline tests/torch/test_hybrid_converter.py:232]: thx for this

File: tests/torch/test_hybrid_converter.py [hunk 5]
```diff
@@ -229,3 +227,14 @@ def test_gpt2_hybrid_mlp_module_not_found():
     fake_module_name = "does_not_exist"
     with pytest.raises(ValueError, match=f"No module found for name {fake_module_name}"):
         HybridFHEModel(model, module_names=fake_module_name)
+
```
Comment (bcm-at-zama) [inline tests/torch/test_hybrid_converter.py:230]: do we somewhere have tests with brevitas models, since we say it works for both torch and brevitas?

**Review by bcm-at-zama — COMMENTED** [review]
Some comments, but take them with a grain of salt, it's been a long time I haven't looked at that precisely

File: src/concrete/ml/torch/hybrid_model.py (hunk 3 above)
Comment (kcelia) [reply src/concrete/ml/torch/hybrid_model.py:365]: > is the if not isinstance(model, torch.nn.Module): valid for Brevitas case as well

Yes, because, our Brevitas models inherit from torch.nn.Module

Comment (kcelia) [reply src/concrete/ml/torch/hybrid_model.py:365]: updated the msg : "The model must be a PyTorch model." to "The model must be a PyTorch or Brevitas model."

File: tests/torch/test_hybrid_converter.py (hunk 5 above)
Comment (kcelia) [reply tests/torch/test_hybrid_converter.py:230]: We have : test_hybrid_brevitas_qat_model

File: docs/guides/hybrid-models.md (hunk 1 above)
Comment (kcelia) [reply docs/guides/hybrid-models.md:144]: We call the quantized private modules, the modes are: simulate, disable, execute and calibrate.

 or really the simulate/fhe function?

Comment (bcm-at-zama) [reply docs/guides/hybrid-models.md:144]: I thought that the simulate/fhe functions were used when we called `run`, which was below `forward` in the non-hybrid models, and that calling simply without the `run` was using the "all-in-CML (ie, no CP, no simulation nor FHE" quantized module function. But I may be wrong, to be checked with the rest of ML team maybe

Comment (kcelia) [reply docs/guides/hybrid-models.md:144]: TBH, I don't know

Comment (jfrery) [reply docs/guides/hybrid-models.md:144]: > here, do we call the quantized module or really the simulate/fhe function?

By default, it's the quantized module. I think it's better to add the forward in the hybrid model as said in this issue https://github.com/zama-ai/concrete-ml-internal/issues/4579.

**Review by andrei-stoian-zama — APPROVED** [review]
